### ISSUE TITLE
waveSharp bug/error : Save configuration to $XDG_CONFIG_HOME

### ISSUE THREAD

**Issue by nouspiro** [CorBer/waveSharp#7]
**Describe the bug**
A clear and concise description of what the bug/error is.

**To Reproduce**
Steps to reproduce the behavior from the start of waveSharp.
1. start waveSharp

**Expected behavior**
That config files are saved into proper location and not clutter $HOME

**Operating system and version**
 - Linux
 - Version 0.1R

**Additional context**
[More information about XDG](https://wiki.archlinux.org/title/XDG_Base_Directory)

**Comment by CorBer** [comment]
Hi,

Can you share a bit more details ? I am using Linux myself as the default OS. For me the files all end up at /home/USERNAME/waveSharp 

cheers
Cor

**Comment by nouspiro** [comment]
Yes that is exactly problem. Config files should be stored in path specified in $XDG_CONFIG_HOME enviroment variable or if not defined then fallback to `$HOME/.config` and not hardcoded path directly in user home. I don't know which framework you use but Qt have method to query paths for various uses https://doc.qt.io/qt-5/qstandardpaths.html

BTW Just tried Windows and that is also just create same folder `C:\Users\name\waveSharp`. On Windows it should be %APPDATA%

Users are not happy with dot folders in theirs home. Folder without dot is even worse. https://www.reddit.com/r/linux/comments/971m0z/im_tired_of_folders_littering_my_home_directory/

**Comment by CorBer** [comment]
Hi,

The "must be" is not a mandatory part of freeware design I think. 
I am not using QT but using freepascal for this application. I might make this an optional feature (so users can decide to store in either way) if I am sure this works on WIN/MAC/LINUX all in the same fashion.  For that I will see this as a request to enhance the application with this choice. Thats not a problem and if this can be created easily it will be in the next update.

cheers
Cor
